code/code/game/objects/items/devices/auto_cpr.dm

```diff
@@ -0,0 +1,417 @@
+/obj/item/auto_cpr
+	name = "stabilizer harness"
+	desc = "A specialized medical harness that gives regular compressions to the patient's ribcage for cases of urgent heart issues, and functions as an emergency \
+	artificial respirator for cases of urgent lung issues."
+	icon = 'icons/obj/med_harness.dmi'
+	icon_state = "med_harness"
+	item_state = "med_harness"
+	w_class = ITEM_SIZE_NORMAL
+	origin_tech = list(TECH_MAGNET = 5, TECH_BIO = 3)
+	slot_flags = SLOT_OCLOTHING
+	var/panel_open = FALSE
+	var/last_pump
+
+	var/obj/item/cell/battery = null
+	var/battery_level = 0
+	var/charge_cost = 7 // 7 for 1 active function. Total of 14 if both CPR and EPP are doing their thing
+	var/cpr_mode = TRUE // Auto-Compressor system. Can be toggled on/off
+
+	var/mob/living/carbon/human/breather = null
+	var/obj/item/clothing/mask/breath/breath_mask = null
+	var/obj/item/tank/tank = null
+	var/tank_level = 0
+	var/tank_type = null
+	var/mask_on = FALSE
+	#define EPP "Emergency Positive Pressure system"
+	var/epp_mode = FALSE // Emergency Positive Pressure system. Can be toggled on/off
+	var/epp_active = FALSE
+
+/obj/item/auto_cpr/Initialize()
+	. = ..()
+	battery = new /obj/item/cell/high(src)
+	breath_mask = new /obj/item/clothing/mask/breath/medical(src)
+	tank = new /obj/item/tank/oxygen_emergency_extended(src)
+	update_icon()
+
+/obj/item/auto_cpr/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	QDEL_NULL(battery)
+	if(breather)
+		breath_mask_off()
+	QDEL_NULL(breath_mask)
+	QDEL_NULL(tank)
+	return ..()
+
+/obj/item/auto_cpr/update_icon()
+	item_state = "[cpr_mode ? "med_harness_cpr" : "[initial(item_state)]"]"
+
+	if(breath_mask)
+		AddOverlays("mask_[mask_on ? "worn" : "idle"]")
```

Comment:
all usage of this needs to be fixed to add_overlay

Reply:
its still not fixed you did add_overlays instead of add_overlay

Reply:
Yes I know, life has gotten busy and I am working on fixing other issues I found before amending the commit